cylc/uiserver/scripts/gui.py

```diff
@@ -20,8 +20,107 @@
 For a multi-user system see `cylc hub`.
 """
 
+import asyncio
+import contextlib
+import json
+import os
+import sys
+
+from pathlib import Path
+from psutil import pid_exists
+
+from cylc.flow.id_cli import parse_id_async
+from cylc.flow.exceptions import InputError
+
 from cylc.uiserver.app import CylcUIServer
 
 
 def main(*argv):
-    return CylcUIServer.launch_instance(argv or None)
+    new_gui = False
+    for arg in argv:
+        if arg == "--new":
+            new_gui = True
+            argv = tuple(x for x in argv if x != "--new")
+            sys.argv.remove("--new")
```

Comment:
Could be simpler. And, dangerous to change list `argv` during iteration over `argv`?
```suggestion
    new_gui = "--new" in argv
    if new_gui:
        argv = tuple(x for x in argv if x != "--new")
        sys.argv.remove("--new")
```